unified-codes/apps/web/src/components/explorer/ExplorerToggleBar.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import { batch, connect } from 'react-redux';
+
+import { EEntityType } from '@unified-codes/data';
+import { CheckCircleIcon, AddIcon, EntityToggleBar } from '@unified-codes/ui';
+
+import ExplorerToggleButton from './ExplorerToggleButton';
+
+import { ExplorerActions, IExplorerAction } from '../../actions';
+import { ExplorerSelectors } from '../../selectors';
+import { withStyles } from '../../styles';
+import { IState, ITheme } from '../../types';
+
+const styles = (_: ITheme) => ({});
+
+const mergeProps = ((stateProps: any, dispatchProps: any) => {
+  const { filterByDrug, filterByMedicinalProduct, filterByOther } = stateProps;
+  const { onToggleFilterByDrug, onToggleFilterByMedicinalProduct, onToggleFilterByOther } = dispatchProps;
```

Comment:
Logic here is that since this is an app component with a specific use, the types should be bound directly to those set by the store, rather than "dynamic". IMO, by making them explicit, it makes the logic simpler and reduces the chances of bugs. By passing types around we might save a few lines of code (although I think it could end up actually as more!), but I think it would be at the cost of clarity and robustness, and I'm not sure what it would give us since this component isn't designed to be used anywhere else?

Reply:
oh all right then. I get what you say about specific usage and will control my compulsion to spot patterns and repetition and try to consolidate things.